driver/src/main/java/org/neo4j/driver/internal/net/pooling/SocketConnectionPool.java

```diff
@@ -68,26 +71,41 @@ public SocketConnectionPool( PoolSettings poolSettings, Connector connector, Clo
     public Connection acquire( final BoltServerAddress address )
     {
         assertNotClosed();
+        BlockingPooledConnectionQueue connectionQueue = pool( address );
+        PooledConnection connection = acquireConnection( address, connectionQueue );
```

Comment:
Not related to this PR but wonder if it would be better API if we let the pooled connection to hold a reference to connecitonValidator and let the pooledConnection to provide methods to test isConnected or isReusable.